go/ql/lib/semmle/go/security/UnicodeBypassValidationQuery.qll

```diff
@@ -0,0 +1,176 @@
+/**
+ * Provides a taint-tracking configuration for detecting "Unicode transformation mishandling" vulnerabilities.
+ */
+
+private import go
+import semmle.go.Concepts
+import semmle.go.dataflow.DataFlow
+import semmle.go.dataflow.TaintTracking
+import UnicodeBypassValidationCustomizations::UnicodeBypassValidation
+
+/** A state signifying that a logical validation has not been performed. */
+class PreValidation extends DataFlow::FlowState {
+  PreValidation() { this = "PreValidation" }
+}
+
+/** A state signifying that a logical validation has been performed. */
+class PostValidation extends DataFlow::FlowState {
+  PostValidation() { this = "PostValidation" }
+}
+
+private predicate indexCheck(DataFlow::Node g, Expr e, boolean outcome) {
+  exists(DataFlow::CallNode cn, DataFlow::EqualityTestNode etn |
+    g = etn and
+    DataFlow::localFlow(cn.getResult(), etn.getAnOperand()) and
+    cn.getTarget()
+        .hasQualifiedName("strings",
+          [
+            "Index", "IndexAny", "IndexByte", "IndexFunc", "IndexRune", "LastIndex", "LastIndexAny",
+            "LastIndexByte", "LastIndexFunc",
+          ]) and
+    cn.getArgument(0).asExpr() = e and
+    etn.getAnOperand().getIntValue() = -1 and
+    outcome = etn.getPolarity()
+  )
+}
+
+private predicate countCheck(DataFlow::Node g, Expr e, boolean outcome) {
+  exists(DataFlow::CallNode cn, DataFlow::EqualityTestNode etn |
```

Comment:
You might also want to consider `RelationalComparisonNode` to pick up cases like Count(...) > 0